core/src/main/java/com/linecorp/armeria/server/healthcheck/HealthChecker.java

```diff
@@ -105,6 +105,20 @@ static HealthChecker ofCpu(double targetSystemCpuUsage, double targetProcessCpuU
         return new CpuHealthChecker(targetSystemCpuUsage, targetProcessCpuUsage);
     }
 
+    /**
+     * Creates a new instance of {@link HealthChecker}
+     * which reports health based on disk space of specific file path.
+     * Both free disk space and total disk space must be (inclusive) lower than the specified threshold in order
+     * for the {@link HealthChecker} to report as healthy.
+     *
+     * @param targetFreeDiskSpace Target free disk space in bytes.
+     * @param targetTotalDiskSpace Target total disk space in bytes.
+     * @return an instance of {@link HealthChecker} configured with the provided disk memory space targets.
+     */
+    static HealthChecker ofDisk(long targetFreeDiskSpace, long targetTotalDiskSpace, String path) {
```

Comment:
I'm not sure it would be very useful to allow users to specify the path.
What do you think of just setting this value as a `new File(".")` like done in spring-boot by default?

ref: https://github.com/spring-projects/spring-boot/blob/ca6ba2e7cf1355fe4764e6bec93dc78f5e76c07e/spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/MetricsProperties.java#L205

I also don't think accepting `targetTotalDiskSpace` would be very useful since it would be mostly static in most cases.

Reply:
Ideally, we should discover all disks in the system automatically, detect the size of the filesystem. We should also allow a user to specify percentage values.

Reply:
I see, I prefer a single health checker is responsible for a single disk as the condition for health checking may become complex. I guess it's fine to leave the `File` parameter as is then.

Reply:
Yeah, we could indeed make use of composition of multiple health checkers to check multiple filesystems. Shall we focus on handling a single filesystem first and see how we could evolve this? That is, yeah, let's keep the `File` parameter.

Reply:
Thanks for all comments :) I am not sure what I should do exactly.
Should I remove percentage values if then?

Reply:
```suggestion
    static HealthChecker ofDisk(long targetFreeDiskSpace, File file) {
```
Then we only need`targetFreeDiskSpace` and `file` as parameter?

Reply:
@injae-kim Thank you for your clarification. I get it. Will fix it within this weekend🙂